src/app/app.component.ts

```diff
@@ -119,11 +124,21 @@ export class AppComponent implements OnInit {
   ngOnInit() {
     const tok = this.helper.decodeToken(this.helper.tokenGetter());
     this.email = tok.email;
+  
 
     // Automatically logout the user after token expiration
     const timeout = tok.exp * 1000 - Date.now();
     setTimeout(() => this.doLogout(), timeout);
 
+    this.accesscode = this.route.snapshot.params['accesscode'];
+    if( this.accesscode ){
+      this.newAccessCodeForm.setValue({access_code: this.accesscode});
+      this.setAccessCode(this.accesscode);
+      this.saveAccessCode();
+      this.doHome();
+      this.accessCodeLinkSuccessAlert = `${this.accesscode} added`;
```

Comment:
This is not executed anymore after you call doHome, or is it?

Reply:
Oh, I didn't pay attention to this dead code. I will take care of it.